src/Umbraco.Cms.Persistence.SqlServer/Services/SqlServerBulkSqlInsertProvider.cs

```diff
@@ -1,74 +1,84 @@
-        private int BulkInsertRecordsSqlServer<T>(IUmbracoDatabase database, PocoData pocoData, IEnumerable<T> records)
+        // create command against the original database.Connection
+        using (DbCommand? command = database.CreateCommand(database.Connection, CommandType.Text, string.Empty))
         {
-            // TODO: The main reason this exists is because the NPoco InsertBulk method doesn't return the number of items.
-            // It is worth investigating the performance of this vs NPoco's because we use a custom BulkDataReader
-            // which in theory should be more efficient than NPocos way of building up an in-memory DataTable.
+            // use typed connection and transaction or SqlBulkCopy
+            SqlConnection tConnection = NPocoDatabaseExtensions.GetTypedConnection<SqlConnection>(database.Connection);
+            SqlTransaction tTransaction =
+                NPocoDatabaseExtensions.GetTypedTransaction<SqlTransaction>(command.Transaction);
+            var tableName = pocoData.TableInfo.TableName;
 
-            // create command against the original database.Connection
-            using (var command = database.CreateCommand(database.Connection, CommandType.Text, string.Empty))
+            if (database.SqlContext.SqlSyntax is not SqlServerSyntaxProvider syntax)
             {
-                // use typed connection and transaction or SqlBulkCopy
-                var tConnection = NPocoDatabaseExtensions.GetTypedConnection<SqlConnection>(database.Connection);
-                var tTransaction = NPocoDatabaseExtensions.GetTypedTransaction<SqlTransaction>(command.Transaction);
-                var tableName = pocoData.TableInfo.TableName;
-
-                var syntax = database.SqlContext.SqlSyntax as SqlServerSyntaxProvider;
-                if (syntax == null) throw new NotSupportedException("SqlSyntax must be SqlServerSyntaxProvider.");
+                throw new NotSupportedException("SqlSyntax must be SqlServerSyntaxProvider.");
+            }
 
-                using (var copy = new SqlBulkCopy(tConnection, SqlBulkCopyOptions.Default, tTransaction)
-                {
-                    BulkCopyTimeout = 0, // 0 = no bulk copy timeout. If a timeout occurs it will be an connection/command timeout.
-                    DestinationTableName = tableName,
-                    // be consistent with NPoco: https://github.com/schotime/NPoco/blob/5117a55fde57547e928246c044fd40bd00b2d7d1/src/NPoco.SqlServer/SqlBulkCopyHelper.cs#L50
-                    BatchSize = 4096
-                })
-                using (var bulkReader = new PocoDataDataReader<T, SqlServerSyntaxProvider>(records, pocoData, syntax))
+            using (var copy = new SqlBulkCopy(tConnection, SqlBulkCopyOptions.Default, tTransaction)
+            {
+                BulkCopyTimeout =
+                           0, // 0 = no bulk copy timeout. If a timeout occurs it will be an connection/command timeout.
```

Comment:
```suggestion
                // 0 = no bulk copy timeout. If a timeout occurs it will be an connection/command timeout.
                BulkCopyTimeout = 0, 
```